CarInspection/src/WebCarInspection/Components/GetDriver.cs

```diff
@@ -0,0 +1,40 @@
+﻿using AutoMapper;
+using BusinessLayer.Ecxeptions;
+using BusinessLayer.Entities;
+using BusinessLayer.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using WebCarInspection.Models;
+
+namespace WebCarInspection.Components
+{
+    public class GetDriver : ViewComponent
+    {
+        private readonly IService<Driver, int> _driverService;
+        private readonly IMapper _mapper;
+
+        public GetDriver(IService<Driver, int> driverService,
+            IMapper mapper)
+        {
+            _driverService = driverService;
+            _mapper = mapper;
+        }
+
+        public async Task<IViewComponentResult> InvokeAsync(int id)
+        {
+            try
+            {
+                var driver = await _driverService.GetByIdAsync(id);
+                var data = _mapper.Map<DriverViewModel>(driver);
+
+                return View(data);
+            }
+            catch (NotFoundException)
+            {
+                var data = new DriverViewModel();
```

Comment:
Can you not send view model at all in this case?

Reply:
Related to all components.

Reply:
I am returning an empty object so that it can be seen that no data was found. When trying to return null, an exception

Reply:
That's strange.
What exception did you get (full stacktrace and description)?
I guess you should fetch null on view side and it will a solution.